src/gatsby-node.js

```diff
@@ -1,35 +1,132 @@
 const {
-        normalizeFields, 
         normalizeEntry, 
         processContentType, 
-        processEntry
+        processEntry,
+        processAsset,
+        makeEntryNodeUid,
+        makeAssetNodeUid
     } = require("./normalize");
 
 const fetchData = require("./fetch");
 
-exports.sourceNodes = async ({ boundActionCreators, createNodeId },configOptions) => {
-    const { createNode } = boundActionCreators;
 
-    const { contentstackData } = await fetchData(configOptions);
+exports.sourceNodes = async ({ actions, getNode, getNodes, createNodeId, store, reporter },configOptions) => {
+    const { createNode, deleteNode, touchNode, setPluginStatus } = actions;
+    let syncToken 
+
+    if (
+        store.getState().status.plugins &&
+        store.getState().status.plugins[`gatsby-source-contentstack`] &&
+        store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ]
+    ) {
+        syncToken = store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ];
+    }
 
+    configOptions.syncToken = syncToken || null;
```

Comment:
Would this typically be passed in via the config? Seems like this should be something computed by the plugin, right?

Reply:
We have computed and just added on configOptions obj and passed in fetch method.